oauth2_http/java/com/google/auth/oauth2/ImpersonatedCredentials.java

```diff
@@ -213,6 +296,89 @@ int getLifetime() {
         ImmutableMap.of("delegates", this.delegates));
   }
 
+  /**
+   * Returns impersonation account credentials defined by JSON using the format generated by gCloud.
+   * The source credentials in the JSON should be either user account credentials or service account
+   * credentials.
+   *
+   * @param json a map from the JSON representing the credentials
+   * @param transportFactory HTTP transport factory, creates the transport used to get access tokens
+   * @return the credentials defined by the JSON
+   * @throws IOException if the credential cannot be created from the JSON.
+   */
+  static ImpersonatedCredentials fromJson(
+      Map<String, Object> json, HttpTransportFactory transportFactory) throws IOException {
+
+    checkNotNull(json);
+    checkNotNull(transportFactory);
+
+    List<String> delegates = null;
+    Map<String, Object> sourceCredentialsJson;
+    String sourceCredentialsType;
+    String quotaProjectId;
+    String targetPrincipal;
+    try {
+      String serviceAccountImpersonationUrl =
+          (String) json.get("service_account_impersonation_url");
+      if (json.containsKey("delegates")) {
+        delegates = (List<String>) json.get("delegates");
+      }
+      sourceCredentialsJson = (Map<String, Object>) json.get("source_credentials");
+      sourceCredentialsType = (String) sourceCredentialsJson.get("type");
+      quotaProjectId = (String) json.get("quota_project_id");
+      targetPrincipal = extractTargetPrincipal(serviceAccountImpersonationUrl);
+    } catch (ClassCastException | NullPointerException | IllegalArgumentException e) {
+      throw new CredentialFormatException("An invalid input stream was provided.", e);
+    }
+
+    GoogleCredentials sourceCredentials;
+    if (GoogleCredentials.USER_FILE_TYPE.equals(sourceCredentialsType)) {
+      sourceCredentials = UserCredentials.fromJson(sourceCredentialsJson, transportFactory);
+    } else if (GoogleCredentials.SERVICE_ACCOUNT_FILE_TYPE.equals(sourceCredentialsType)) {
+      sourceCredentials =
+          ServiceAccountCredentials.fromJson(sourceCredentialsJson, transportFactory);
+    } else {
+      throw new IOException(
+          String.format(
+              "A credential of type %s is not supported as source credential for impersonation.",
+              sourceCredentialsType));
+    }
+    return ImpersonatedCredentials.newBuilder()
+        .setSourceCredentials(sourceCredentials)
+        .setTargetPrincipal(targetPrincipal)
+        .setDelegates(delegates)
+        .setScopes(new ArrayList<String>())
+        .setLifetime(DEFAULT_LIFETIME_IN_SECONDS)
+        .setHttpTransportFactory(transportFactory)
+        .setQuotaProjectId(quotaProjectId)
+        .build();
+  }
+
+  @Override
+  public boolean createScopedRequired() {
```

Comment:
This method is not invented in this PR. It is defined in the base class 'GoogleCredentials' which is overridden by its subclasses. I think it uses "create" in its name because of its relationship with the createScoped method. 

https://github.com/googleapis/google-auth-library-java/blob/292e81a93242c76e3b45abad331bef57f8c02ce0/oauth2_http/java/com/google/auth/oauth2/GoogleCredentials.java#L218